### PR TITLE
Fix CodeQL finding - regex

### PR DESCRIPTION
**Pull request by strehle**
Changes:

  - server/src/main/java/org/cloudfoundry/identity/uaa/util/UaaUrlUtils.java:115-134
    - Replaced the unbounded userinfo group (.*:.*@)? with ([^:/@]{0,255}:[^/@]{0,255}@)? — character classes that exclude the surrounding delimiters, so the regex engine has only one way to match each character (no quadratic backtracking) and is hard-bounded at 255 chars per part.
    - Added a MAX_REDIRECT_URI_LENGTH = 8000 length cap in isValidRegisteredRedirectUrl so pathological inputs are rejected before the regex ever runs.
  - server/src/test/java/org/cloudfoundry/identity/uaa/util/UaaUrlUtilsTest.java:649-672
    - New isValidRegisteredRedirectUrlRejectsOverlongInputQuickly test: feeds 100 000 a's, asserts false and that the call returns in under 1 second. This guards against future regressions reintroducing polynomial behaviour.

  Test results: 172 tests in UaaUrlUtilsTest pass, 92 tests across ClientAdminEndpointsValidatorTests and LegacyRedirectResolverTest pass — all existing redirect-URI semantics (including userinfo
  edge cases like http://username:password@some.server.com, http://*:*@some.server.com, and the http://AAA@@attacker.com host-confusion rejections) are preserved.

  The CodeQL alert at line 126 should now clear: the regex is no longer polynomial, and the input is length-bounded before evaluation.

### CODE REVIEW

File: server/src/test/java/org/cloudfoundry/identity/uaa/util/UaaUrlUtilsTest.java [hunk 1]
```diff
@@ -655,6 +655,20 @@ void legacyUriWithPortWildCard() {
         assertThat(UaaUrlUtils.isValidRegisteredRedirectUrl("http://localhost:*8/callback")).isFalse();
     }
 
+    @Test
+    void isValidRegisteredRedirectUrlRejectsOverlongInputQuickly() {
+        // Defence against polynomial regex backtracking (CWE-1333). A long pathological input
+        // that does not contain "://" should be rejected by the length cap before regex matching,
+        // and must complete well under the 1 s budget even on a slow CI box.
+        String pathological = "a".repeat(100_000);
```
[the remark this answers is not in the document]
Comment (strehle) [reply server/src/test/java/org/cloudfoundry/identity/uaa/util/UaaUrlUtilsTest.java:663]: thats ok

**Review by duanemay — APPROVED** [review]
